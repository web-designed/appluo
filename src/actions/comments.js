
import moment from 'moment'
import database from '../firebase/firebase'

//*******************************************************
// COMMENTS ACTION GENERATORS
//*******************************************************

   export const addComment = (eventId, comment) => ({
      type: 'ADD_COMMENT',
      eventId,
      comment
   })

   export const startAddComment = (eventId, {commentBody, commenter}) => {
      return(dispatch) => {
         const comment = {
            createdAt: moment().valueOf(),
            commenter,
            commentBody
         }

         return database.ref(`events/${eventId}/comments`).push(comment).then( ref => {
            dispatch(addComment(eventId, {
               ...comment,
               id: ref.key
            }))
         })
      }
   }

   export const removeComment = (eventId, commentId) => ({
      type:'REMOVE_COMMENT',
      eventId,
      commentId
   })

   export const editComment = (eventId, commentId, update) => ({
      type:'EDIT_COMMENT',
      eventId,
      commentId,
      update
   })