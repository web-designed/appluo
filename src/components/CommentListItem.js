import React from 'react'

const CommentListItem = ({ commentsId, createdAt, commenter, commentBody }) => {

   return (
      <div>
         <h3>{commentBody}</h3>
         <p>{commenter} - {createdAt} </p>
      </div>
   )
}

export default CommentListItem