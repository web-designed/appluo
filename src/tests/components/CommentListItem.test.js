import React from 'react'
import { shallow } from 'enzyme'
import CommentListItem from '../../components/CommentListItem'
import comments from '../fixtures/comments'

test('Should render CommentListItem correctly', () => {
   const wrapper = shallow(<CommentListItem />)
   expect(wrapper).toMatchSnapshot()
})