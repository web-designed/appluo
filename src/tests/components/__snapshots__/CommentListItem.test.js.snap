// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render CommentListItem correctly 1`] = `""`;
